### DISCUSSION TITLE
applying non-continuous hard boundary condition

### DISCUSSION THREAD

**Discussion by bmemlh** [lululxvi/deepxde#1820] [Q&A]
Hello, everyone!

I am trying to perform a 2D amperometric electrochemical model by utilizing hard-boundary conditions

I have defined only one chemical species and its concentration has neumann boundary conditions on left, right and upper boundaries. On the other hand, the lower boundary is partitioned. While r<0.1, a dirichlet boundary condition with the concentration value of 0.2529 is applied upon the electrode boundary, and neumann boundary condition is valid for the rest of the lower boundary. Moreover, I have an initial condition as C=1.

I have defined the hard boundary conditions as below:

```
def output_transform(x, C):
r, z, t = x[:, 0:1], x[:, 1:2], x[:, 2:3]
C_new = tf.where(tf.logical_and(r<=0.1, z==0), tf.ones_like(C)*0.25292364, t*((r*(1-r))**2+(z*(4-z))**2)*C+1)
return C_new
```

When I executed the code, all collocation and boundary points are always calculated as unity; thus, 0.2529 concentration value would never be predicted on the electrode boundary

I have schematically summarized the boundary/initial conditions with the picture below

![boundary-initial conditions](https://github.com/user-attachments/assets/024a249b-a432-41a1-884e-af5dc4af74c7)

Can you help me to solve this problem?
Thank you in advance!

**Comment by lululxvi** [comment]
This is difficult. It might be easier to use soft constraint.

**Reply by bmemlh** [reply]
Thank you, Lu Lu!

Yours sincerely
